src/cryptonote_core/service_node_list.cpp

```diff
@@ -417,16 +417,23 @@ namespace service_nodes
     auto iter = m_service_nodes_infos.find(key);
     if (iter != m_service_nodes_infos.end())
     {
-      service_node_info &old_info = iter->second;
-      uint64_t expiry_height = old_info.registration_height + get_staking_requirement_lock_blocks(m_blockchain.nettype());
+      int hard_fork_version = m_blockchain.get_hard_fork_version(block_height);
+      if (hard_fork_version >= cryptonote::Blockchain::version_10_swarms)
+      {
+        service_node_info &old_info = iter->second;
+        uint64_t expiry_height = old_info.registration_height + get_staking_requirement_lock_blocks(m_blockchain.nettype());
+        if (block_height < expiry_height)
+          return;
 
-      if (block_height < expiry_height)
+        // NOTE: Node preserves its position in list if it reregisters during grace period.
+        registered_during_grace_period = true;
+        info.last_reward_block_height = old_info.last_reward_block_height;
+        info.last_reward_transaction_index = old_info.last_reward_transaction_index;
```

Comment:
Info gets filled out in is_registration_tx(), which sets the registration height to block_height. Only thing we need to inherit is the reward height/index since that's what we use to order the list.